# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/SocketProcessBridgeParent.cpp
# Commit: c414d82a5325
# Full Hash: c414d82a532508523d40ce63111cb30b2a1076be
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-02-22 05:26:22
# Regressor Bug: 1521879
# File Overlap Count: 1
# Description:
#   Bug 1521879 - Part 1: IPC-based MediaTransport implementation r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D17273
# ==============================================================================

diff -r 3a182b05e2c1 -r c414d82a5325 netwerk/ipc/SocketProcessBridgeParent.cpp
--- a/netwerk/ipc/SocketProcessBridgeParent.cpp	Thu Feb 21 19:01:32 2019 +0200
+++ b/netwerk/ipc/SocketProcessBridgeParent.cpp	Thu Feb 21 16:42:12 2019 +0000
@@ -8,6 +8,9 @@
 
 #include "mozilla/ipc/BackgroundParent.h"
 #include "SocketProcessChild.h"
+#ifdef MOZ_WEBRTC
+#include "signaling/src/peerconnection/MediaTransportHandlerParent.h"
+#endif
 
 namespace mozilla {
 namespace net {
@@ -56,5 +59,21 @@
   SocketProcessChild::GetSingleton()->DestroySocketProcessBridgeParent(mId);
 }
 
+dom::PMediaTransportParent*
+SocketProcessBridgeParent::AllocPMediaTransportParent() {
+#ifdef MOZ_WEBRTC
+  return new MediaTransportHandlerParent;
+#endif
+  return nullptr;
+}
+
+bool SocketProcessBridgeParent::DeallocPMediaTransportParent(
+    dom::PMediaTransportParent* aActor) {
+#ifdef MOZ_WEBRTC
+  delete aActor;
+#endif
+  return true;
+}
+
 }  // namespace net
 }  // namespace mozilla